octobot_trading/personal_data/orders/order_util.py

```diff
@@ -435,3 +437,55 @@ async def get_order_size_portfolio_percent(exchange_manager, order_amount, side,
+
+
+async def update_from_order_storage(order, exchange_manager, pending_groups):
+    # only real orders can be updated by stored orders
+    if exchange_manager.is_trader_simulated or exchange_manager.is_backtesting \
+            or not exchange_manager.storage_manager.orders_storage:
+        return
+    order_details = await exchange_manager.storage_manager.orders_storage.get_startup_order_details(order.order_id)
+    if order_details:
+        await _update_from_order_details(order, order_details, exchange_manager, pending_groups)
+
+
+async def create_order_from_order_storage_details(order_storage_details, exchange_manager, pending_groups):
+    order = orders_storage.create_order_from_storage_details(
+        exchange_manager.trader,
+        order_storage_details
+    )
+    await _update_from_order_details(order, order_storage_details, exchange_manager, pending_groups)
+    return order
+
+
+async def _update_from_order_details(order, order_details, exchange_manager, pending_groups):
+    # rebind order attributes that are not stored on exchange
+    order_dict = order_details[orders_storage.OrdersStorage.ORIGIN_VALUE_KEY]
+    order.tag = order_dict[enums.ExchangeConstantsOrderColumns.TAG.value]
+    order.trader_creation_kwargs = order_details[enums.PersistedOrdersAttr.TRADER_CREATION_KWARGS.value]
+    order.exchange_creation_params = order_details[enums.PersistedOrdersAttr.EXCHANGE_CREATION_PARAMS.value]
+    order.set_shared_signal_order_id(order_details[enums.PersistedOrdersAttr.SHARED_SIGNAL_ORDER_ID.value])
+    order.has_been_bundled = order_details[enums.PersistedOrdersAttr.HAS_BEEN_BUNDLED.value]
+    order.associated_entry_ids = order_details[enums.PersistedOrdersAttr.ENTRIES.value]
+    group = order_details[enums.PersistedOrdersAttr.GROUP.value]
+    if group:
+        group_name = group[enums.PersistedOrdersAttr.GROUP_ID.value]
+        if group_name not in pending_groups:
+            pending_groups[group_name] = exchange_manager.exchange_personal_data.orders_manager.get_or_create_group(
+                group_util.get_group_type(group[enums.PersistedOrdersAttr.GROUP_TYPE.value]),
+                group_name,
+            )
+        order.order_group = pending_groups[group_name]
+    chained_orders = order_details[enums.PersistedOrdersAttr.CHAINED_ORDERS.value]
+    if chained_orders:
+        for chained_order in chained_orders:
+            chained_order_inst = await create_order_from_order_storage_details(
+                chained_order, exchange_manager, pending_groups
+            )
+            await chained_order_inst.set_as_chained_order(
+                order,
+                chained_order[enums.PersistedOrdersAttr.HAS_BEEN_BUNDLED.value],
+                chained_order[enums.PersistedOrdersAttr.EXCHANGE_CREATION_PARAMS.value],
+                **chained_order[enums.PersistedOrdersAttr.TRADER_CREATION_KWARGS.value],
+            )
+            order.add_chained_order(chained_order_inst)
```

Comment:
I think we should put this code somewhere else, I was surprised to find an order creation in a method named `_update_from_order_details`. I think we need a method that updates order from storage data and a second method that creates chained orders if a loaded order from storage had chained orders.